pkg/controllers/networkpolicy/sync.go

```diff
@@ -130,15 +130,21 @@ func (a *appliedToSecurityGroup) sync(syncContent *securitygroup.Synchronization
 
 	// roughly count and compare rules in syncContent against nps.
 	// also updates cloudRuleIndexer in the process.
-	for _, iRule := range syncContent.IngressRules {
-		countIngressRuleItems(&iRule, items, true)
-		if updated := a.checkAndUpdateIndexer(r, &iRule, cloudRuleMap); updated {
+	for _, rule := range syncContent.IngressRules {
+		iRule := rule.Rule.(*securitygroup.IngressRule)
+		if rule.NetworkPolicyName != "" {
+			countIngressRuleItems(iRule, items, true)
+		}
+		if updated := a.checkAndUpdateIndexer(r, &rule, cloudRuleMap); updated {
```

Comment:
Can be changed to 
if a.checkAndUpdateIndexer(r, &rule, cloudRuleMap) {
   indexerUpdate = true
}